infra/ballista/build/Dockerfile

```diff
@@ -0,0 +1,112 @@
+
+ARG RELEASE_FLAG=release
+ARG FUNCTION_DIR="/function"
+
+
+# build ballista components
+FROM rust:1.63.0-buster as builder
+
+# Include global arg in this stage of the build
+ARG RELEASE_FLAG
+
+ENV RUST_LOG=info
+ENV RUST_BACKTRACE=full
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get update && \
+    apt-get -y install \
+    libssl-dev \
+    openssl \
+    zlib1g \
+    zlib1g-dev \
+    libpq-dev \
+    cmake \
+    protobuf-compiler \
+    netcat \
+    curl \
+    unzip \
+    git
+
+# create build directories
+RUN mkdir -p /home/builder/workspace
+
+ENV HOME=/home/builder
+ENV PATH=$HOME/.cargo/bin:$PATH
+
+# prepare rust
+RUN rustup update && \
+    rustup component add rustfmt && \
+    cargo install cargo-chef --version 0.1.34
+
+WORKDIR /home/builder/workspace/
+
+# RUN git clone https://github.com/apache/arrow-ballista.git
+COPY arrow-ballista /home/builder/workspace/arrow-ballista
+
+WORKDIR /home/builder/workspace/arrow-ballista
+
+RUN ls -l ../
+
+# Actually build the project
+RUN cargo build --features "flight-sql s3" --${RELEASE_FLAG}
+
+
+FROM debian:11.4 as ric-dependency
+
+# Include global arg in this stage of the build
+ARG RELEASE_FLAG
+ARG FUNCTION_DIR
+
+ENV RELEASE_FLAG=${RELEASE_FLAG}
+ENV RUST_LOG=info
+ENV RUST_BACKTRACE=full
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Install aws-lambda-cpp build dependencies
+RUN apt-get update && \
+    apt-get -y install\
+    g++ \
+    make \
+    cmake \
+    unzip \
+    python3 \
+    python3-pip \
+    netcat \
+    libcurl4-openssl-dev
+
+# Include global arg in this stage of the build
+ARG FUNCTION_DIR
+# Create function directory
+RUN mkdir -p ${FUNCTION_DIR}
+
+# Copy function code
+COPY lambdatization/infra/ballista/build/lambda-handler.py ${FUNCTION_DIR}
+
+# Install the runtime interface client and lambda requirements
+RUN pip3 install \
+    --target ${FUNCTION_DIR} \
+    awslambdaric pexpect
+
+# copy built scheduler into accesible folder
+COPY --from=builder /home/builder/workspace/arrow-ballista/target/$RELEASE_FLAG/ballista-scheduler /opt/ballista/ballista-scheduler
+
+# Expose Ballista Scheduler gRPC port
+EXPOSE 50050
+
+RUN mkdir -p /tmp/scheduler/sled
+
+# copy built executor into accesible folder
+COPY --from=builder /home/builder/workspace/arrow-ballista/target/$RELEASE_FLAG/ballista-executor /opt/ballista/ballista-executor
+
+# Expose Ballista Executor gRPC port
+EXPOSE 50051
+
+RUN mkdir -p /tmp/executor
+
+# copy builtcli into accesible folder
+COPY --from=builder /home/builder/workspace/arrow-ballista/target/$RELEASE_FLAG/ballista-cli /opt/ballista/ballista-cli
+
+WORKDIR ${FUNCTION_DIR}
+
+ENTRYPOINT [ "python3", "-m", "awslambdaric" ]
+CMD [ "lambda-handler.handler" ]
```

Comment:
❓  This is not separating the final image from the one that builds and installs `awslambdaric` anymore. This means that the final image will have all the `lambdaric` build dependencies installed. I wouldn't mark this as critical as it is just an optimization, but as it was present in previous versions I am wondering why it was removed

Reply:
because merging the images was the only solution I found for the cryptic error you mentioned above:

> * ❓ didn't you get a `.terraform.lock.hcl` file when you deployed the lambda?
> * ❓ is `l12n ballista.lambda-example` working for you? I get a very weird `Runtime.ExitError`  with a very cryptic stack trace:
> 
> ```
> File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
> return _run_code(code, main_globals, None,
> File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
> exec(code, run_globals)
> File "/function/awslambdaric/__main__.py", line 25, in <module>
> main(sys.argv)
> File "/function/awslambdaric/__main__.py", line 21, in main
> bootstrap.run(app_root, handler, lambda_runtime_api_addr)
> File "/function/awslambdaric/bootstrap.py", line 399, in run
> event_request = lambda_runtime_client.wait_next_invocation()
> File "/function/awslambdaric/lambda_runtime_client.py", line 68, in wait_next_invocation
> response_body, headers = runtime_client.next()
> ```